### PR TITLE
docs: rework readme

### PR DESCRIPTION
**Pull request by noah-eigenfeld**
#### What do these changes do/fix?
Breaks down and refreshes our readme instructions, per some customer feedback we got a few months back. Implementation for https://github.ibm.com/Watson-Discovery/disco-issue-tracker/issues/14221

#### How do you test/verify these changes?
Open the READMEs in this branch, and verify that everything renders correctly
#### Have you documented your changes (if necessary)?
N/A
#### Are there any breaking changes included in this pull request?
No

### CODE REVIEW

File: packages/discovery-react-components/README.md [hunk 1]
```diff
@@ -20,6 +26,235 @@
...
+
+3. Wholesale CSS
+
+```html
+<link rel="stylesheet" href="/path/to/discoverey-components/index.css" />
+```
+
+This method uses the vanilla CSS that is built from the Discovery Component styles SCSS files and also includes the Carbon Component styles. Use this method if you do not have SCSS importing as part of your application build pipeline.
+
+<!-- TODO: do we need this section as well? -->
```
Comment (noah-eigenfeld) [inline packages/discovery-react-components/README.md:256]: Delete TODO now that this section has been moved to the components sub-folder

File: README.md [hunk 2]
```diff
@@ -48,365 +45,17 @@ First, you will need to customize and improve your document retrieval project on
 
 - [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
 - [nvm](https://github.com/nvm-sh/nvm#installation-and-update)
-  - Set Node version to 16.x.x via `nvm install 16` and `nvm use 16`
+  - Set Node version to 16.x.x via `nvm install 16` and `nvm use 16` (also compatible with Node 18)
```
Comment (jhpedemonte) [inline README.md:48]: Did you verify that it works as expected with Node 18?

File: README.md [hunk 3]
```diff
@@ -35,11 +28,15 @@
     - [Feature tests](#feature-tests)
     - [Continuous integration](#continuous-integration)
   - [Branching and Releasing](#branching-and-releasing)
-- [Helpful links](#helpful-links)
-- [Contributors](#contributors)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
+## Quick Links
```
Comment (jhpedemonte) [inline README.md:34]: Just noticed we don't have an "about" statement, describing what this repo/code is for. I suggest adding an `About` section above this one and writing a short description. Or maybe is should go before `Table of contents`?

File: examples/discovery-search-app/README.md [hunk 4]
```diff
@@ -1,5 +1,156 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+**Table of Contents** _generated with [DocToc](https://github.com/thlorenz/doctoc)_
```
Comment (jhpedemonte) [inline examples/discovery-search-app/README.md:4]: We should remove these comments and the "generated with ..." section, unless we intend to consistently run the generator.

Comment (jhpedemonte) [reply examples/discovery-search-app/README.md:4]: The images in this file don't load. Click on "View file" to see what it looks like rendered.

File: README.md (hunk 2 above)
Comment (noah-eigenfeld) [reply README.md:48]: Yes

File: examples/discovery-search-app/README.md (hunk 4 above)
Comment (noah-eigenfeld) [reply examples/discovery-search-app/README.md:4]: I didn't realize until I committed my changes that doctoc actually runs during the commit hooks. So, in that sense I think that we should keep a mention of DocToc in here, but add to the comments that you don't have to manually run it in order to update the ToC

File: README.md [hunk 5]
```diff
@@ -31,6 +32,10 @@
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
+## About
+
+This is a repository created and maintained by the IBM Watson Discovery UI team. It contains a collection of components, to be used by internal and external applications to query [Watson Discovery](https://www.ibm.com/products/watson-discovery) projects.
```
Comment (jhpedemonte) [inline README.md:37]: Add a line pointing to the storybook. Something like "Take a look at our [Storybook](...) for info on these components."

**Review by jhpedemonte — APPROVED** [review]
Just one additional suggestion. With that fixed, approved.

**Pull request merged**